### PR TITLE
chore(sg): sg bazel uses remote cache by default (unless in CI) 

### PR DESCRIPTION
**Pull request by jhchabran**
Using a GCS bucket + a custom credential helper, we can enable the remote cache for everyone without having to perform explicit authentication or setup. Authentication is fully delegated to `gcloud`. And because every teammate can access the secrets for the local env, they can also write on the cache. 

In case of emergency, `--disable-remote-cache` can disable the remote cache flags, and normal `bazel [cmd]` are unaffected. 

Bystanders that would like to test this: 

```
git fetch
git checkout jh/bzl/local-remote-cache
go run ./dev/sg bazel build //cmd/gitserver
```

💡 Additionally, this creates the incentive we were looking for to teach folks to use `sg bazel` over `bazel`.

## Test plan

Locally tested (and @eseliger tested it too, to avoid ending up in a scenario where I have special perms on that GCP project).

### CODE REVIEW

File: dev/sg/sg_bazel.go [hunk 1]
```diff
@@ -51,13 +51,47 @@ var bazelCommand = &cli.Command{
...
+				disableRemoteCache = true
+			case "--disable-remote-cache=false":
+				disableRemoteCache = false
+			default:
+				args = append(args, arg)
+			}
+		}
+
+		// If we end up running `sg bazel` in CI, we don't want to use the remote cache for local environment,
+		// sowe force disable the flag explicilty.
```
Comment (Strum355) [inline dev/sg/sg_bazel.go:81]:
```suggestion
		// so we force disable the flag explicilty.
```

**Review by Strum355 — APPROVED** [review]
:nice:

File: .aspect/bazelrc/remote_cache_for_local.bazelrc [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+build --remote_download_toplevel
+
+# Upload locally executed action results to the remote cache.
+# Docs: https://bazel.build/reference/command-line-reference#flag--remote_upload_local_results
+build --remote_upload_local_results
+
+# Fall back to standalone local execution strategy if remote execution fails. If the grpc remote
+# cache connection fails, it will fail the build, add this so it falls back to the local cache.
+# Docs: https://bazel.build/reference/command-line-reference#flag--remote_local_fallback
+build --remote_local_fallback
```
Comment (Strum355) [inline .aspect/bazelrc/remote_cache_for_local.bazelrc:12]: Is this for remote cache or remote _execution_? 🤔

Comment (jhchabran) [reply .aspect/bazelrc/remote_cache_for_local.bazelrc:12]: IIRC this applies to remote caching too

**Pull request merged**